prompt_toolkit/completion/word_completer.py

```diff
@@ -31,7 +31,7 @@ class WordCompleter(Completer):
 
     def __init__(
         self,
-        words: Union[List[str], Callable[[], List[str]]],
+        words: Union[Iterable[str], Callable[[], Iterable[str]]],
```

Comment:
Please make this at least a ``Sequence``.

Otherwise, users are going to pass in a generator of strings, and wonder why the second time, they're not getting any completions.

Is there a reason it should be ``Iterable`` here?

Reply:
Thanks for the review @jonathanslenders 

Nope. I was just going to use the lowest possible. You are right a `Sequence` would be more fitting.

I stopped working on this PR as some  other troubles occured (see failing tests)
Once I found time to resolve them, I will undraft the PR and ping you.